app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,13 @@
+class SessionsController < ApplicationController   
+  def create
+    auth = request.env["omniauth.auth"]     
+    user = User.find_by_provider_and_uid(auth["provider"], auth["uid"]) || User.create_with_omniauth(auth)
+    session[:user_id] = user.id
+    redirect_to root_url, :notice => "Signed in!"
```

Comment:
:redirect_to is an inappropriate action here since we are not serving up any views ourselves. We should instead return a success response to the client